File: LeetCode/1431_kids_with_the_greatest_number_of_candies.php
<?php

declare(strict_types=1);

use Solution as GlobalSolution;

/**
 * Template Soultion class for LeetCode problems
 * 
 */
class Solution
{
    /**
     * @param array $candies
     * @param int $extraCandies
     * @return array
     */
    public function kidsWithCandies($candies, $extraCandies)
    {
        $kids = [];

        foreach ($candies as $kid) {
            if ($kid + $extraCandies >= max($candies)) {
                $kids[] = true;
            } else {
                $kids[] = false;
            }
        }
        return $kids;
    }
}
// test suite
$solution = new GlobalSolution();
var_dump($solution->kidsWithCandies([2, 3, 5, 1, 3], 3));
var_dump($solution->kidsWithCandies([4, 2, 1, 1, 2], 1));

File: LeetCode/1431_kids_with_the_greatest_number_of_candies.py
#!/usr/bin/env python3
# pylint: disable=invalid-name, too-few-public-methods, unused-import, no-self-use,missing-function-docstring,consider-using-enumerate
'''
    LeetCode solution class
'''
from typing import List, Optional


class Solution:
    '''
    LeetCode solution class
    '''

    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        return [True if kid + extraCandies >=
                max(candies) else False for kid in candies]


solution = Solution()
# [true,true,true,false,true]
print(solution.kidsWithCandies(candies=[2, 3, 5, 1, 3], extraCandies=3))

# [true,false,false,false,false]
print(solution.kidsWithCandies(candies=[4, 2, 1, 1, 2], extraCandies=1))
